### PR TITLE
Cast jsonarray to list

### PR DESCRIPTION
**Pull request by sophiaso**
refs #22

### CODE REVIEW

File: src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java [hunk 1]
```diff
@@ -137,54 +131,47 @@ public void testFindByIdPartition() {
...
+        assertTrue(result.get(0).equals(person2));
     }
 
     @Test
     public void testDeleteByIdPartition() {
         setupPartition();
 
         // insert new document with same partition key
-        final Person person2 = new Person("newid", "newfn", TEST_PERSON.getLastName());
+        final Person person2 = new Person("newid", "newfn", TEST_PERSON.getLastName(), null);
```
Comment (yungez) [inline src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java]: why null

File: src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java [hunk 2]
```diff
@@ -137,54 +131,47 @@ public void testFindByIdPartition() {
...
-        assertThat(result.getId()).isEqualTo(updated.getId());
-        assertThat(result.getFirstName()).isEqualTo(updated.getFirstName());
-        assertThat(result.getLastName()).isEqualTo(updated.getLastName());
+        assertTrue(result.equals(updated));
     }
 
     @Test
     public void testDeleteById() {
-        final Person person2 = new Person("newid", "newfn", "newln");
+        final Person person2 = new Person("newid", "newfn", "newln", null);
```
Comment (yungez) [inline src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java]: why null

File: src/main/java/com/microsoft/azure/spring/data/documentdb/repository/DocumentDbRepository.java [hunk 3]
```diff
@@ -6,8 +6,8 @@
 
 package com.microsoft.azure.spring.data.documentdb.repository;
 
-import org.springframework.data.repository.NoRepositoryBean;
 import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.NoRepositoryBean;
```
Comment (yungez) [inline src/main/java/com/microsoft/azure/spring/data/documentdb/repository/DocumentDbRepository.java]: remove

File: src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java [hunk 4]
```diff
@@ -24,19 +24,21 @@
...
 @RunWith(SpringJUnit4ClassRunner.class)
 @PropertySource(value = {"classpath:application.properties"})
 public class DocumentDbTemplateIT {
-
     private static final String TEST_ID = "testid";
     private static final String TEST_NOTEXIST_ID = "testid2";
 
     private static final String TEST_DB_NAME = "testdb";
-    private static final Person TEST_PERSON = new Person(TEST_ID, "testfirstname", "testlastname");
+    private static final List<String> hobbies = Arrays.asList("photography", "fishing");
```
Comment (yungez) [inline src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java]: could you pls try to test with nested object array to find out where's gap to support nested object?

File: src/main/java/com/microsoft/azure/spring/data/documentdb/repository/DocumentDbRepository.java (hunk 3 above)
Comment (sophiaso) [reply src/main/java/com/microsoft/azure/spring/data/documentdb/repository/DocumentDbRepository.java]: This is necessary, I used intellij optimized the import, so the import order changed a little bit.

File: src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java (hunk 4 above)
Comment (sophiaso) [reply src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java]: I will do a local test first.

File: src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java (hunk 1 above)
Comment (sophiaso) [reply src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java]: Will change to some other more meaningful value.

File: src/main/java/com/microsoft/azure/spring/data/documentdb/core/convert/MappingDocumentDbConverter.java [hunk 5]
```diff
@@ -64,12 +67,37 @@ public MappingDocumentDbConverter(
...
+                }
+
+                final Object fieldValue = sourceDocument.get(prop.getName());
+
+                if (fieldValue instanceof JSONArray) {
+                    final List<Object> mappedList = new ArrayList<>();
+                    for (int i = 0; i < ((JSONArray) fieldValue).length(); i++) {
+                        Object value;
+                        if (prop.getActualType().isPrimitive() || prop.getActualType().equals(String.class)) {
+                            value = ((JSONArray) fieldValue).getString(i);
```
Comment (yungez) [inline src/main/java/com/microsoft/azure/spring/data/documentdb/core/convert/MappingDocumentDbConverter.java]: interesting :)

**Pull request merged**